storage/Devices/DeviceImpl.h

```diff
@@ -21,6 +21,32 @@ namespace storage
     class SystemInfo;
 
 
+    template <typename Type>
+    Type* to_device_of_type(Device* device)
+    {
+	static_assert(!is_const<Type>::value, "Type must not be const");
+
+	Type* tmp = dynamic_cast<Type*>(device);
+	if (!tmp)
+	    ST_THROW(DeviceHasWrongType("device has wrong type"));
```

Comment:
To be honest, I expect such kind of exception constuct message at it own. Something like
```
ST_THROW(DeviceHasWrongType(Type, device))
```
and all that typeid, getting classname should be hidden inside DeviceHasWrongType constructor. just my 2c